### PR TITLE
Feature/golem stats improvements

### PR DESCRIPTION
**Pull request by grisha87**
This PR implements improvements identified when building Golem Stats 2 website.

### CODE REVIEW

File: src/market/scan/scanned-offer.ts [hunk 1]
```diff
@@ -51,34 +56,104 @@ export class ScannedOffer {
...
+   * }
+   * ```
+   */
+  get paymentPlatformAddresses(): PaymentPlatformAddressSet {
+    const platformProps = Object.entries(this.model.properties).filter(([key]) =>
+      key.startsWith("golem.com.payment.platform."),
+    );
+
+    const platformAddress = platformProps.map(([key, address]) => [
+      key.replace("golem.com.payment.platform.", "").replace(".address", ""),
```
Comment (SewerynKras) [inline src/market/scan/scanned-offer.ts:153]: if someone sets a constraint that starts with `golem.com.payment.platform` but ends with something different from `.<platform>.address` then this method will produce nonsense. We should filter the properties not only with `startsWith` but also with `endsWith` or even better, with a single regex like
`/^golem\.com\.payment\.platform\.[a-zA-Z\-0-9]+\.address$/` (untested)

**Review by SewerynKras — CHANGES_REQUESTED** [review]

File: src/market/scan/scanned-offer.ts [hunk 2]
```diff
@@ -145,14 +145,15 @@ export class ScannedOffer {
...
-    const platformProps = Object.entries(this.model.properties).filter(([key]) =>
-      key.startsWith("golem.com.payment.platform."),
-    );
-
-    const platformAddress = platformProps.map(([key, address]) => [
-      key.replace("golem.com.payment.platform.", "").replace(".address", ""),
-      address,
-    ]);
+    const propRegex = /golem.com.payment.platform.([a-z0-9-]+).address/;
+    const platformProps = Object.entries(this.model.properties).filter(([key]) => key.match(propRegex));
```
Comment (SewerynKras) [inline src/market/scan/scanned-offer.ts:149]: Is this first filter needed anymore? You're filtering again based on the same condition below

File: src/market/scan/scanned-offer.ts [hunk 3]
```diff
@@ -145,14 +145,15 @@ export class ScannedOffer {
...
   get paymentPlatformAddresses(): PaymentPlatformAddressSet {
-    const platformProps = Object.entries(this.model.properties).filter(([key]) =>
-      key.startsWith("golem.com.payment.platform."),
-    );
-
-    const platformAddress = platformProps.map(([key, address]) => [
-      key.replace("golem.com.payment.platform.", "").replace(".address", ""),
-      address,
-    ]);
+    const propRegex = /golem.com.payment.platform.([a-z0-9-]+).address/;
```
Comment (SewerynKras) [inline src/market/scan/scanned-offer.ts:148]: 1. `.` should be escaped, right now it matches any character (try `golem☠com☠payment☠platform☠nonsense☠address`)
2. are we sure all constraints keys are lowercase only?

Comment (grisha87) [reply src/market/scan/scanned-offer.ts:148]: AD 1. Will fix that.
AD 2. Yes, that's the case.

File: src/market/scan/scanned-offer.ts (hunk 2 above)
Comment (grisha87) [reply src/market/scan/scanned-offer.ts:149]: As per our call: It's not needed, I will drop it.

**Review by SewerynKras — APPROVED** [review]

**Pull request merged**